### PR TITLE
Adds support for multidate, daterange and multidaterange to csv import

### LINKED ISSUE #4903: Multi date range and multi date ranges are not supported by CSV import
The property type date range and multi date range are not supported through csv import. At present, this type of information has to be manually entered into the database as the import leaves out these property types when trying to do an import. 

User Story: 
As an admin user responsible for the migration of a large dataset of existing data into Uwazi that contains crucial date range information, I would like to be able to import this information through the csv file as well instead of having to manually enter this into each entity after the import is completed.

### PR DESCRIPTION
**Pull request by fnocetti**
fixes #4903 

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

File: app/api/csv/specs/typeParsers.spec.js [hunk 1]
```diff
@@ -110,4 +110,120 @@ describe('csvLoader typeParsers', () => {
       }
     );
   });
+
+  describe('multidate', () => {
+    it.each`
+      dateProp                                 | dateFormat      | expectedDate
```
Comment (LaszloKecskes) [inline app/api/csv/specs/typeParsers.spec.js:116]: Maybe also test: 
'' (empty string)
'01-01-2014'
'01-01-2014||01-01-2015' (double separator in the middle)
'2014|2015|' (terminate with separator)

File: app/api/csv/specs/typeParsers.spec.js [hunk 2]
```diff
@@ -110,4 +110,120 @@ describe('csvLoader typeParsers', () => {
...
+          expectedDate
+        );
+      }
+    );
+  });
+
+  describe('daterange', () => {
+    it.each`
+      dateProp                                 | dateFormat      | expectedDate
+      ${'2014:2015'}                           | ${'dd/MM/yyyy'} | ${{ from: '01-01-2014', to: '01-01-2015' }}
```
Comment (LaszloKecskes) [inline app/api/csv/specs/typeParsers.spec.js:147]: Similarly:
'' (empty string)
':2015' 
'2014:'

File: app/api/csv/specs/typeParsers.spec.js [hunk 3]
```diff
@@ -110,4 +110,120 @@ describe('csvLoader typeParsers', () => {
...
+          to: moment.utc(expected[0].value.to, 'X').format('DD-MM-YYYY'),
+        }).toEqual(expectedDate);
+      }
+    );
+  });
+
+  describe('multidaterange', () => {
+    it.each([
+      {
+        dateProp: '2014:2015|2016:2017',
```
Comment (LaszloKecskes) [inline app/api/csv/specs/typeParsers.spec.js:174]: If you think it is worth it, the same extra cases as in the multidate case.

File: app/api/csv/specs/typeParsers.spec.js [hunk 4]
```diff
@@ -110,4 +110,120 @@ describe('csvLoader typeParsers', () => {
       }
     );
   });
+
+  describe('multidate', () => {
+    it.each`
+      dateProp                                 | dateFormat      | expectedDate
+      ${'2014|2015'}                           | ${'dd/MM/yyyy'} | ${['01-01-2014', '01-01-2015']}
+      ${'2014 11 6|2015 11 6'}                 | ${'dd/MM/yyyy'} | ${['06-11-2014', '06-11-2015']}
```
Comment (LaszloKecskes) [inline app/api/csv/specs/typeParsers.spec.js:118]: Should here the format not be 'dd MM yyyy', or is it that flexible? This is for a couple of other ones as well.

Comment (LaszloKecskes) [reply app/api/csv/specs/typeParsers.spec.js:118]: Scratch that, I see the implementation. :)

**Review by LaszloKecskes — CHANGES_REQUESTED** [review]
I've left a couple of comment to indicate a few possible test cases. We did a few of these for the multiselects (see arrangeThesauriTest.csv). Aside from that, the code looks good, and works.

File: app/api/csv/specs/typeParsers.spec.js (hunk 1 above)
Comment (fnocetti) [reply app/api/csv/specs/typeParsers.spec.js:116]: Added some extra cases.

File: app/api/csv/specs/typeParsers.spec.js (hunk 3 above)
Comment (fnocetti) [reply app/api/csv/specs/typeParsers.spec.js:174]: Same as multidate

**Review by LaszloKecskes — APPROVED** [review]
Looks good!

**Pull request merged**